spark-operator/src/main/java/org/apache/spark/k8s/operator/utils/PodUtils.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.k8s.operator.utils;
+
+import static org.apache.spark.k8s.operator.utils.ModelUtils.findDriverMainContainerStatus;
+
+import java.util.List;
+
+import io.fabric8.kubernetes.api.model.ContainerStatus;
+import io.fabric8.kubernetes.api.model.Pod;
+
+import org.apache.spark.k8s.operator.spec.ApplicationSpec;
+
+public class PodUtils {
+
+  public static boolean isPodReady(final Pod pod) {
+    if (pod == null
+        || pod.getStatus() == null
+        || pod.getStatus().getConditions() == null
+        || pod.getStatus().getConditions().isEmpty()) {
+      return false;
+    }
+    return pod.getStatus().getConditions().parallelStream()
+        .anyMatch(
+            condition ->
+                "ready".equalsIgnoreCase(condition.getType())
```

Comment:
Although this is a well-known string, please use `PodPhase` and its methods instead of this kind of direct string.